pyro/poutine/subsample_messenger.py

```diff
@@ -132,7 +133,7 @@ def _process_message(self, msg):
         msg["scale"] = msg["scale"] * self.size / self.subsample_size
 
     def _postprocess_message(self, msg):
-        if msg["type"] == "param" and self.dim is not None:
+        if msg["type"] in ("param", "subsample") and self.dim is not None:
```

Comment:
The error message in 145-147 will be incorrect for `subsample`s because they have no `msg["name"]`

Reply:
fixed.